Question_Categorization/cluster_or_tag.py

```diff
@@ -0,0 +1,92 @@
+import sys
+sys.path.append('../')
+from nltk.corpus.reader import tagged
+from Bagyasree.Testing.testing import Test
+from Question_Categorization.clustering_and_tagging_classes import Clustering, Tagging, General
+
+class FinalAlgorithms:
+
+    def get_question_strings(self, questions):
+        try:
+            #Create arrays of question IDs and question strings with matching indices.
+            question_ids = []
+            question_strings = []
+            for question in questions:
+                question_ids.append(question['id'])
+                question_strings.append(question['text'])
+            
+            return question_ids, question_strings
+        
+        except Exception as e:
+            print('Error: ',e)
+            raise e
+
+    #Agglomerative clustering using Universal Sentence Encoding
+    def clustering(self, questions):
+
+        try:
+
+            question_ids, question_strings = self.get_question_strings(questions)
+
+            #Algorithm on question strings.
+            vectors = General().universal_sentence_encoder(question_strings)
+            cluster_mapping = Clustering().agglomerative_clustering(vectors)
+
+            #Match question IDs to clusters.
+            clusters= {}
+            for i in range(0, len(cluster_mapping)):
```

Comment:
I've added that to the clustering function, because it's a lot cleaner, but the tagging function still requires the IDs, because it receives a dictionary.